library/billrep.inc

```diff
@@ -154,27 +154,23 @@ function getBillsListBetween(
     "order by billing.pid, billing.date ASC";
 
     $res = sqlStatement($sql, array($code_type));
-    $string = "( ";
+    $array = array();
     for ($iter=0; $row=sqlFetchArray($res); $iter++) {
-        $string .= $row{"id"}.",";
+        array_push($array, $row{"id"});
     }
 
-    $string = substr($string, 0, strlen($string)-1);
-    $string .= ")";
-    return $string;
+    return $array;
 }
 
-function billCodesList($list, $skip = "()")
+function billCodesList($list, $skip = [])
 {
-    if ($list == "()") {
+    if (empty($list)) {
         return;
     }
 
-    if ($skip == "()") {
-        sqlStatement("update billing set billed=1 where id in $list");
-    } else {
-        sqlStatement("update billing set billed=1 where id in $list and id not in $skip");
-    }
+    $sqlBindArray = array_diff($list, $skip);
```

Comment:
rec putting another check for empty $list here in case they all went away on the array_diff:
```
    if (empty($list)) {
        return;
    }
```